internal/pkg/service/stream/storage/node/coordinator/fileimport/operator.go

```diff
@@ -237,10 +237,17 @@ func (o *operator) importFile(ctx context.Context, file *fileData) {
 
 	var err error
 	if !file.IsEmpty {
-		// Import the file to specific provider
-		err = o.plugins.ImportFile(ctx, &file.File)
+		// Get file statistics
+		var stats statistics.Aggregated
+		stats, err = o.statistics.FileStats(ctx, file.FileKey)
 		if err != nil {
-			err = errors.PrefixError(err, "error when waiting for file import")
+			err = errors.PrefixError(err, "cannot get file statistics")
+		} else {
+			// Import the file to specific provider
+			err = o.plugins.ImportFile(ctx, &file.File, stats.Staging)
+			if err != nil {
+				err = errors.PrefixError(err, "error when waiting for file import")
+			}
```

Comment:
This error handling is quite error-prone, it would be better to wrap it in some method and use `return err`.

Reply:
So 

```
	var err error
	if !file.IsEmpty {
	  ....
	}
```

May be:
```
	err := o.doImportFile(....)
```

Reply:
Maybe also `dbCtx` and `SwitchToImported` may be part of the method.

So the `if` condition before the `IncrementRetryAttempt` will cover all errors.

`dbCtx` there would be 2x, but that doesn't seem like a problem to me.

Reply:
The `successfully imported file` may be closer to the `SwitchToImported`, in `if/else`.